container/utils.go

```
@@ -6,20 +6,39 @@ package container
 import (
 	"os/exec"
 	"runtime"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/core/instance"
 )
 
-var RunningInContainer = func() bool {
-	if runtime.GOOS != "linux" {
-		return false
+func pathForExcutable(executableName string) (string, error) {
+	var ePath string
+	if path, err := exec.LookPath(executableName); err == nil {
```

err is not handled completely